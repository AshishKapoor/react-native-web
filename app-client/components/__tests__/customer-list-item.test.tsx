import React from "react";
import renderer, { ReactTestRendererJSON, create } from "react-test-renderer";
import CustomerListItem from "../customer-list-item";

import { TamaguiProvider } from "tamagui";
import config from "../../tamagui.config";

const customer = {
  __typename: "ZellerCustomer",
  id: 8,
  name: "Harry Potter",
  role: "Admin",
  email: "test8@test.com",
};

describe("Components/CustomerListItem", () => {
  const CustomerListItemComponent = (
    <TamaguiProvider config={config}>
      <CustomerListItem customer={customer} />
    </TamaguiProvider>
  );

  it("renders correctly", () => {
    const customerListItem = create(
      CustomerListItemComponent
    ).toJSON() as ReactTestRendererJSON;
    expect(customerListItem?.type).toBe("View");
  });

  it("matches snapshot", () => {
    const tree = renderer
      .create(<CustomerListItem customer={customer} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
