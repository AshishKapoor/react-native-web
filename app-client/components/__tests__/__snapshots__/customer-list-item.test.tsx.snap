// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/CustomerListItem matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "row",
      "marginBottom": 32,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "hsl(209, 100%, 96.5%)",
        "borderBottomLeftRadius": 9,
        "borderBottomRightRadius": 9,
        "borderTopLeftRadius": 9,
        "borderTopRightRadius": 9,
        "flexDirection": "column",
        "height": 44,
        "justifyContent": "center",
        "maxHeight": 44,
        "maxWidth": 44,
        "minHeight": 44,
        "minWidth": 44,
        "width": 44,
      }
    }
  >
    <Text
      cursor="default"
      fontFamily="$body"
      id=":r3:"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "color": "#4169e1",
          "display": "flex",
          "fontFamily": "Inter",
          "fontSize": 14,
          "letterSpacing": 0,
          "lineHeight": 23,
        }
      }
      suppressHighlighting={true}
      userSelect="none"
    >
      H
    </Text>
  </View>
  <View
    pointerEvents="none"
    style={
      {
        "alignItems": "stretch",
        "height": 18,
        "minHeight": 18,
        "minWidth": 18,
        "width": 18,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
      }
    }
  >
    <Text
      cursor="default"
      fontFamily="$body"
      id=":r4:"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "color": "$color",
          "display": "flex",
          "fontFamily": "Inter",
          "fontSize": 14,
          "letterSpacing": 0,
          "lineHeight": 23,
        }
      }
      suppressHighlighting={true}
      userSelect="none"
    >
      Harry Potter
    </Text>
    <Text
      cursor="default"
      fontFamily="$body"
      id=":r5:"
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "color": "grey",
          "display": "flex",
          "fontFamily": "Inter",
          "fontSize": 14,
          "letterSpacing": 0,
          "lineHeight": 23,
        }
      }
      suppressHighlighting={true}
      userSelect="none"
    >
      Admin
    </Text>
  </View>
</View>
`;
